CMakeLists.txt

```diff
@@ -140,7 +140,7 @@ option(USE_METAL "Use Metal for iOS build" ON)
 option(USE_NATIVE_ARCH "Use -march=native" OFF)
 cmake_dependent_option(
     USE_NCCL "Use NCCL" ON
-    "USE_CUDA;UNIX;NOT APPLE" OFF)
+    "USE_CUDA OR USE_ROCM;UNIX;NOT APPLE" OFF)
```

Comment:
Does this work for you? It wasn't working for me, that's why @xuhdev suggested using the `USE_GPU` method.

Reply:
It shows USE_NCCL has been turned on in the CI https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py3.6-clang7-rocmdeb-ubuntu16.04-build/47/consoleFull:

```
18:48:09 --   USE_NCCL              : ON
18:48:09 --     USE_SYSTEM_NCCL     : ON
```

so I believe it's working.

Reply:
I thought I suggested `USE_GPU` was merely for cleaner code? The previous PR used a lot of `USE_CUDA OR USE_ROCM` across the codebase, if I remember correctly.

Reply:
Okay, my bad, I must be misremembering. I agree it seems to be working in the ROCm runs on the CI, so I have no issues with landing this PR if everything seems okay!

Reply:
@xuhdev Agree USE_GPU would be cleaner. I think there are more places using the `USE_CUDA OR USE_ROCM` logic (other than the places that touch nccl). Maybe let's do the cleanup across all the places in a separate diff?

Reply:
Well, if you do want to go down that path, you might then want to take another look at my original PR 23884, since that decision forced a couple of other changes .... might avoid reinventing the wheel.

Reply:
That being said, I'd much rather have it in a separate PR in the interest of expediency for this PR.

Reply:
Sure, let's do this after this PR is merged